docs/src/get-started/concepts.rst

```diff
@@ -0,0 +1,239 @@
+store it's own data.
+
+.. py:currentmodule:: equistore
+
+.. _concept-TensorMap:
+
+TensorMap
+^^^^^^^^^
+
+The core type of equistore is the :py:class:`TensorMap`: a high dimensional
+block-sparse tensor containing both data and metadata. A TensorMap contains a
+list of block (represented as :ref:`concept-TensorBlock`), each associated with
+a key. The set of all keys is stored in a :ref:`concept-Labels` object
+(described below), which can be accessed in Python via
+:py:attr:`TensorMap.keys`.
+
+
+The keys can contain multiple dimensions (in the illustration below we have two
+dimensions named ``lambda`` and ``sigma``), and each entry in the keys has one
+integer value for each dimension. Here for example, the first block is
+associated with ``lambda = 0`` and ``sigma = 0``, while the second block is
+associated with ``lambda = 0`` and ``sigma = 1``, and so on.
+
+.. figure:: ../../static/images/TensorMap.*
+    :width: 600px
+    :align: center
+
+    Illustration of an equistore TensorMap object, made of a set of keys and
+    associated :ref:`concept-TensorBlock`.
+
+
+Different key dimensions can have different purposes, but some typical keys
+dimensions you'll encounter when working with atomistic data are the following:
+
+- **atomic species dimensions**: when using a one-hot encoding of different
+  atomic species (or atomic elements) in a structure, the resulting data is
+  sparse, containing implicit zeros if a given species is not present in a
+  structure. This is the case of the ``species_center`` and various
+  ``species_neighbor`` key dimensions produced by `rascaline`_.
+- **symmetry markers**: Another use case for equistore is to store and
+  manipulate equivariant data, i.e. data that transforms in a known, specific
+  way when the corresponding atomic structure is transformed. This is typically
+  used to represent the symmetry property of some data with respect to
+  rotations, by decomposing the properties of interest on a basis of spherical
+  harmonics. When handling this kind of data, it is convenient to store and
+  manipulate the data corresponding to different spherical harmonics (or
+  generally different irreducible representations of the symmetry group)
+  separately. This is the case of the ``spherical_harmonics_l`` key dimension
+  produced by `rascaline`_: different blocks will contain the :math:`\ell = 1`
+  and :math:`\ell = 2` parts of an equivariant representation.
+
+.. _rascaline: https://github.com/Luthaf/rascaline/
+
+.. _concept-Labels:
+
+Labels
+^^^^^^
+
+A fundamental part of equistore is to carry simultaneously the data used in
+machine learning and the associated metadata. The first kind of metadata we
+encountered was the keys of a :py:class:`TensorMap`, stored as an instance of
+the :py:class:`Labels` class. This class is also used to store all other
+metadata in equistore, i.e. all the metadata associated with a given
+:py:class:`TensorBlock`.
```

Comment:
```suggestion
machine learning and the associated metadata. :py:class:`Labels` 
serves as a superclass for many forms of metadata, including
our :py:class:`TensorMap` keys.
```

Reply:
Labels is not a superclass (i.e. parent class w.r.t. inheritance), it is the class/type which all metadata is an instance of